agent/src/agent/data_extractor/cacti/repository.py

```diff
@@ -10,98 +10,136 @@ def _get_session(connection_string: str):
+            AND data_source_path IS NOT NULL
+            AND gti.graph_type_id IN (4, 5, 6, 7 ,8);
+        """)
+        for row in res:
+            local_graph_id = row['local_graph_id']
+            if local_graph_id not in self.graphs:
+                self.graphs[local_graph_id] = {}
+            if 'items' not in self.graphs[local_graph_id]:
+                self.graphs[local_graph_id]['items'] = {}
+            self.graphs[local_graph_id]['items'][row['item_id']] = {'data_source_path': row['data_source_path']}
+
+    def _get_graph_titles(self):
+        res = self.session.execute("""
+            SELECT gtg.local_graph_id, gtg.title, gl.host_id
+            FROM graph_templates_graph gtg
+            JOIN graph_local gl ON gtg.local_graph_id = gl.id
+            WHERE local_graph_id != 0
+        """)
+        for row in res:
+            if row['local_graph_id'] not in self.graphs:
+                continue
+            self.graphs[row['local_graph_id']]['title'] = row['title']
+            self.graphs[row['local_graph_id']]['host_id'] = str(row['host_id'])
+
+    def _get_graph_variables(self):
+        res = self.session.execute("""
+            select gl.id, field_name, field_value
+            from host_snmp_cache hsc
+            join graph_local gl
+            on gl.host_id = hsc.host_id
+            and gl.snmp_index = hsc.snmp_index
+            and gl.snmp_query_id = hsc.snmp_query_id
+        """)
+        for row in res:
+            local_graph_id = row['id']
+            if local_graph_id not in self.graphs:
+                continue
+            if 'variables' not in self.graphs[local_graph_id]:
+                self.graphs[local_graph_id]['variables'] = {}
+            self.graphs[local_graph_id]['variables'][row['field_name']] = row['field_value']
+
+    def _get_graph_items(self):
+        # (4, 5, 6, 7 ,8) are ids of AREA, STACK, LINE1, LINE2, and LINE3 graph item types
+        res = self.session.execute("""
+            SELECT gti.id as item_id, gti.local_graph_id, dtr.data_source_name, gti.text_format as item_title
+            FROM graph_templates_item gti
+            JOIN graph_templates_graph gtg on gti.local_graph_id = gtg.local_graph_id
+            JOIN data_template_rrd dtr on dtr.id = gti.task_item_id
+            WHERE gtg.local_graph_id != 0
+            AND gti.graph_type_id IN (4, 5, 6, 7 ,8)
+        """)
+        for row in res:
+            local_graph_id = row['local_graph_id']
+            if local_graph_id not in self.graphs:
+                continue
+            if 'items' not in self.graphs[local_graph_id]:
+                self.graphs[local_graph_id]['items'] = []
+            if row['item_id'] not in self.graphs[local_graph_id]['items']:
+                continue
+
+            self.graphs[local_graph_id]['items'][row['item_id']]['data_source_name'] = row['data_source_name']
+            self.graphs[local_graph_id]['items'][row['item_id']]['item_title'] = row['item_title']
```

Comment:
_get_sources query and this one are the same, maybe we can join data_local here and get the rrd path?